From 1a9f015ef18954985610dfe0a233f2d2fdf18392 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Sun, 2 Oct 2016 17:26:06 +0200
Subject: [PATCH] etcetera: Add fractional Etc/GMT time zones

:Release Notes:
Add fractional Etc/GMT time zones. The format changed in 2016g release
so we should check if settings app really depend on GMT* format or can
use new numeric formats as well.

:Detailed Notes:
It was introduced as a .patch in this meta-lg-webos change:
  commit 68c9a2a2f71938cc0a1a369462b8e11734ab1de4
  Author: Sapna Todwal <sapna.todwal@lge.com>
  Date:   Tue May 27 15:15:23 2014 -0700

    starfish-image: tzdata=starfish1

    :Release Notes:
    Add fractional Etc/GMT time zones.

    :Detailed Notes:
    The settings app now needs to be able to add manual timezones which
    can be anywhere from GMT-12, GMT-11:30, GMT-11 ... GMT+13:30, GMT+14.
    So this change adds all these new fractional timezones to the
    /usr/share/zoneinfo/Etc directory.

    :Testing Perfomed:
    Verified that the /usr/share/zoneinfo/Etc directory now lists all the new
    fractional timezones.

The format was changed in upstream change:
https://github.com/eggert/tz/commit/5659c5322976ea171e6a5afe14e9bc8172a51e24
  Author: Paul Eggert <eggert@cs.ucla.edu>
  Date:   Tue Aug 23 16:35:24 2016 -0700
  Subject: Phase out alphanumeric abbrs from etcetera

And again in 2024b:
https://github.com/eggert/tz/commit/94e6b3b0eb773e66a9ac49b905c69c0fa6a31dc3
  Author: Paul Eggert <eggert@cs.ucla.edu>
  Date:   Mon Apr 8 13:11:19 2024 -0700
  Subject Switch to %z in main dataform

    * NEWS: Mention this.
    * africa, antarctica, asia, australasia, etcetera, europe:
    * northamerica, southamerica: Use %z for time zone abbreviations,
    except in rearguard sections.
    * ziguard.awk: Adjust code to reflect the fact that %z is now
    in the main dataform.

:Testing Performed:

:QA Notes:

:Issues Addressed:
---
Upstream-Status: Pending

 etcetera | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/etcetera b/etcetera
index a5ecd6de..31896786 100644
--- a/etcetera
+++ b/etcetera
@@ -52,28 +52,54 @@ Link	Etc/GMT				GMT
 # Also, the time zone abbreviations are now compatible with %z.
 
 Zone	Etc/GMT-14	14	-	%z
+Zone	Etc/GMT-13:30	13:30	-	%z
 Zone	Etc/GMT-13	13	-	%z
+Zone	Etc/GMT-12:30	12:30	-	%z
 Zone	Etc/GMT-12	12	-	%z
+Zone	Etc/GMT-11:30	11:30	-	%z
 Zone	Etc/GMT-11	11	-	%z
+Zone	Etc/GMT-10:30	10:30	-	%z
 Zone	Etc/GMT-10	10	-	%z
+Zone	Etc/GMT-9:30	9:30	-	%z
 Zone	Etc/GMT-9	9	-	%z
+Zone	Etc/GMT-8:30	8:30	-	%z
 Zone	Etc/GMT-8	8	-	%z
+Zone	Etc/GMT-7:30	7:30	-	%z
 Zone	Etc/GMT-7	7	-	%z
+Zone	Etc/GMT-6:30	6:30	-	%z
 Zone	Etc/GMT-6	6	-	%z
+Zone	Etc/GMT-5:30	5:30	-	%z
 Zone	Etc/GMT-5	5	-	%z
+Zone	Etc/GMT-4:30	4:30	-	%z
 Zone	Etc/GMT-4	4	-	%z
+Zone	Etc/GMT-3:30	3:30	-	%z
 Zone	Etc/GMT-3	3	-	%z
+Zone	Etc/GMT-2:30	2:30	-	%z
 Zone	Etc/GMT-2	2	-	%z
+Zone	Etc/GMT-1:30	1:30	-	%z
 Zone	Etc/GMT-1	1	-	%z
+Zone	Etc/GMT-0:30	0:30	-	%z
+Zone	Etc/GMT+0:30	-0:30	-	%z
 Zone	Etc/GMT+1	-1	-	%z
+Zone	Etc/GMT+1:30	-1:30	-	%z
 Zone	Etc/GMT+2	-2	-	%z
+Zone	Etc/GMT+2:30	-2:30	-	%z
 Zone	Etc/GMT+3	-3	-	%z
+Zone	Etc/GMT+3:30	-3:30	-	%z
 Zone	Etc/GMT+4	-4	-	%z
+Zone	Etc/GMT+4:30	-4:30	-	%z
 Zone	Etc/GMT+5	-5	-	%z
+Zone	Etc/GMT+5:30	-5:30	-	%z
 Zone	Etc/GMT+6	-6	-	%z
+Zone	Etc/GMT+6:30	-6:30	-	%z
 Zone	Etc/GMT+7	-7	-	%z
+Zone	Etc/GMT+7:30	-7:30	-	%z
 Zone	Etc/GMT+8	-8	-	%z
+Zone	Etc/GMT+8:30	-8:30	-	%z
 Zone	Etc/GMT+9	-9	-	%z
+Zone	Etc/GMT+9:30	-9:30	-	%z
 Zone	Etc/GMT+10	-10	-	%z
+Zone	Etc/GMT+10:30	-10:30	-	%z
 Zone	Etc/GMT+11	-11	-	%z
+Zone	Etc/GMT+11:30	-11:30	-	%z
 Zone	Etc/GMT+12	-12	-	%z
